test/state/host.cpp

```diff
@@ -493,7 +493,7 @@ evmc_tx_context Host::get_tx_context() const noexcept
         m_block.prev_randao,
         0x01_bytes32,  // Chain ID is expected to be 1.
         uint256be{m_block.base_fee},
-        intx::be::store<uint256be>(m_block.blob_base_fee),
+        intx::be::store<uint256be>(compute_blob_gas_price(m_block.excess_blob_gas.value_or(0))),
```

Comment:
I think we should not compute this each access. Maybe we need two sets of fields in `BlockInfo`: official/header fields and computed values.

If big deal, we can handle this by separate PR.

Reply:
Oh right, get_tx_context() is called within the instruction, yeah, better to fix right away :(. Alternatively this can be kept in Host? WDYT? if we can avoid mixing header/computed values, maybe it's better? or is Host just inappropriate for this?

Reply:
Better to put it back to `BlockInfo` because this is also used in transaction validation and transition. These are also important cases.

Reply:
argh, I've done it and only after I recalled from previous week, that I have already been there. The problem which arises is that it makes `compute_blob_gas_price` come before block validation, making problematic inputs (excess_blob_gas) come through, which previously wasn't the case. `fake_exponential` seems to go into infinite loop for such inputs.

I'll need to revisit this and look more closely at what exactly is happening

Reply:
I imagine we keep the newly added field and also have `blob_base_fee` "after" them computed as some point...

Reply:
I explored this idea, but I'm struggling to find a good point to compute that. It would sacrifice some `const`s and make a mess, or force us to introduce more complexity to manage that.

I have a counter-proposal. I dug deeper into `fake_exponential`, and it seems to me like not only it infinite loops but also easily overflows over 256 bits. If I cap the result at uint256::max on mul overflow (I can post code in a moment), things work OK. I think the tacit assumption in `fake_exponential` spec is that it should use arbitrary precision, and I peeked at some impls and they actually do that.

So I'd rather fix `fake_exponential` to not loop/overflow, and rely on current block validation, which would reject the block properly. 

I see three options here:
1. switch from uint256 to big int of some sort
2. detect overflow and cap `compute_blob_gas_price` to uint256::max
3. detect overflow and return nullopt from `compute_blob_gas_price`

2 is the easiest (will push in a sec), but has the disadvantage of there being a block with potentially incorrect blob_base_fee field, hanging around until we reject it (on validation of excess/used blob gas fields).

3 seems better but not sure if the opts won't proliferate around the code base excessively. Maybe not